app/presenters/applicant_account_presenter.rb

```diff
@@ -0,0 +1,21 @@
+class ApplicantAccountPresenter
+  def initialize(applicant_bank = nil)
+    @applicant_bank = applicant_bank
+  end
+
+  def main_account_holder_name
+    @applicant_bank.bank_account_holders.last.full_name
+  end
+
+  def main_account_holder_address
+    @applicant_bank.bank_account_holders.last.addresses.first.values.join(', ')
```

Comment:
That's a very long chain of methods without any checks that they each exist.

Also `@applicant_bank.bank_account_holders.last` is used in a couple of methods. If the definition of `main bank account` changes (to `first` for example), both would need to be updated. I'd suggest

```ruby
def main_account_holder
  @applicant_bank.bank_account_holders.&last
end

def main_account_holder_name
  main_account_holder.&full_name
end

def main_account_holder_address
  main_account_holder.&addresses.&first.&values.&join(', ')
end
```
For the last method - even better may be to have an `Address` presenter:
```ruby
def main_account_holder_address
  address = AddressPresenter.new(main_account_holder.&addresses.&first)
  address.values
end
```

Reply:
Actually, thinking about this a little more: I don't think the definition of "Main account holder" belongs in a presenter. That belongs in the Bank Provider. I'd suggest updating the model:

```ruby
class BankProvider < ApplicationRecord
  ...
  has_one :main_account_holder, class_name: :BankAccountHolder, -> { order: (created_at: :desc) }
```

Reply:
Also, the `BankAccountHolder` model doesn't have an `addresses` field.